XUtils.h

```diff
@@ -117,6 +117,13 @@ int compareRealNumbers(double a, double b);
    nonnegative. */
 double sumPositiveValues(const double* array, size_t count);
 
+/* Returns the number of trailing zero bits */
+#if defined(HAVE_BUILTIN_CTZ)
+#define countTrailingZeros(x) __builtin_ctz(x)
```

Comment:
1. `__builtin_ctz` return type is `int`, not `unsigned int`. Maybe add a cast?
2. `__builtin_ctz` has undefined behavior when `x == 0`. Maybe add an assert() or handle it as a special case.